### PR TITLE
[0.4.x] libvisual-plugins: Fix out-of-bounds access for actor "lv_flower"

### PR DESCRIPTION
**Pull request by hartwork**

### CODE REVIEW

File: libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c [hunk 1]
```diff
@@ -280,14 +280,14 @@ int lv_flower_render (VisPluginData *plugin, VisVideo *video, VisAudio *audio)
 		float y;
 		int i;
 		for (i=0; i<priv->nof_bands; i++) {
-			y=temp_bars[i * 8];
+			y=temp_bars[(i * 8) % NOTCH_BANDS];
```
Comment (kaixiong) [inline libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c:283]: I see that this fixes the overflow. But could this line have been intended to be:
```c
y=temp_bars[i];
```

**Review by kaixiong — COMMENTED** [review]
Nice catch! I have a question though.

File: libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c (hunk 1 above)
Comment (kaixiong) [reply libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c:283]: Or perhaps:

```cpp
y=temp_bars[i] * 8;
```

Comment (hartwork) [reply libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c:283]: @kaixiong good question! I tried `temp_bars[i] * 8;` and then the flower goes even more crazy while its already jumping around a lot so I ruled out that as intended. `temp_bars[i];` was the other candidate.

I checked Git history now and found that:
- `float temp_bars[NOTCH_BANDS];` was first introduced in commit 939266aa5b31d540c367408e812f8d1e2a9b9248.
- The commit replaces `y=freqnorm[i * 8];` by `y=temp_bars[i * 8];`.
- However `freqnorm` is defined as `float freqnorm[256];` was used, which is 8 times bigger.
- Hence `i * 8` likely should have been `i` in that commit, the first place.

What do you think?

Comment (kaixiong) [reply libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c:283]: @hartwork, yeah I guess we will fix it to just `i` instead.

I kind of threw in the last bit about `temp_bars[i] * 8`. The flower already looks correctly scaled, magnifying it by 8 times is nuts to say the least.

Comment (hartwork) [reply libvisual-plugins/plugins/actor/pseudotoad_flower/actor_flower.c]: @kaixiong done

**Review by kaixiong — APPROVED** [review]

**Pull request merged**